src/main/java/com/example/imagescaler/upscale/apiprovider/UpscaleApiProviderImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.example.imagescaler.upscale.apiprovider;
+
+import com.example.imagescaler.upscale.configuration.ApiProperties;
+import java.time.Duration;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+@Service
+public class UpscaleApiProviderImpl implements UpscaleApiProvider {
+  private final ApiProperties apiProperties;
+  private final WebClient webClient;
+
+  public UpscaleApiProviderImpl(
+    ApiProperties apiProperties,
+    WebClient.Builder webClientBuilder
+  ) {
+    this.apiProperties = apiProperties;
+    this.webClient = initializeWebClient(webClientBuilder);
```

Comment:
I don't think it's worth to create another method to initialize the web client. This is the only caller of that method. And initializing it here means that most probably you don't need a field for `ApiProperties`.